### PR TITLE
YhPg8d6a CRUD operations for buses

### PR DESCRIPTION
**Pull request by GuevaraMarius**
### WHAT DOES THIS PR DO?
This PR adds **CRUD** operations for buses 
### DESCRIPTION OF TASK TO BE COMPLETED
The operator should be able to create, read, update, and delete a bus.
The operator should be able to view a paginated list of buses in the system.
The users should be able to view a bus's details.
Appropriate error messages are displayed on unsuccessful attempts.

### HOW SHOULD THIS BE MANUALLY TESTED?
- Do `npm install` 
- Create `.env` file and put configs as in `.env.example`
-Create three databases: `phantom-dev` , `phantom-test`, `phantom` 
-Do `npm run migrate ` 
**Testing using postman**

- Make a POST request to the bus `/api/bus` along with data that involves keys of the brand, plate, and driver and their values to **create** a bus
- Make a GET request to the bus `/api/bus` to **read** all bus's details in the database
- Make a GET request to the bus `/api/bus/id` where id is the unique id given to a specific bus to get that bus's details
- Make a PATCH request to the bus `/api/bus/id`  along with data that involves keys of the brand, plate, and driver and their values to **update** the specific bus with its unique id
- Make a DELETE request to the bus `/api/bus/id` to **delete** that bus with the specific id
### What are relevant trello story
https://trello.com/c/YhPg8d6a/11-crud-operations-for-buses

### CODE REVIEW

**Review by SIBONIYO — APPROVED** [review]

**Review by KundwaBruno — COMMENTED** [review]
Great work!, I can suggest that you stop tracking the .DS_store file which i think it is not needed!

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,4 @@ node_modules
 .nyc_output
 .env
 build
-.DS_STORE
\ No newline at end of file
+.DS_STORE
```
Comment (41hulk) [inline .gitignore]: You made a Typing error
 ```diff 
-.DS_STORE
+.DS_Store
```

File: src/controllers/translate.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//     }
+// }
+
+// detectLanguage('Hello world!')
+//     .then((res) => {
+//         console.log(res);
+//     })
+//     .catch((err) => {
+//         console.log(error);
+//     });
```
Comment (41hulk) [inline src/controllers/translate.js]: You can delete them if they are not in use

**Review by 41hulk — COMMENTED** [review]
Great work Guevara, just some modification 🎢

**Review by SIBONIYO — APPROVED** [review]

File: src/controllers/bus.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      
+    }
+}
+// export const getBus= async (req,res)=>{
+//     try {
+//         const bus=await Bus.findAll();
+//         return successRes(res, 200, 'Buses found', bus);
+
+//     } catch (error) {
+//         return res.status(400).send({
```
Comment (mansurissa) [inline src/controllers/bus.js]: It can be better if you remove commented codes

File: src/controllers/translate.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+//     try {
+//         let response = await translate.detect(text);
+//         return response[0].language;
+//     } catch (error) {
+//         console.log(`Error at detectLanguage --> ${error}`);
+//         return 0;
+//     }
+// }
+
+// detectLanguage('Hello world!')
```
Comment (mansurissa) [inline src/controllers/translate.js]: Same as here

**Review by mansurissa — APPROVED** [review]
Great job @Guevara and Samuel but also remember to create a PR with one commit(you need to swash)

File: src/controllers/bus.js (hunk 3 above)
Comment (GuevaraMarius) [reply src/controllers/bus.js]: Thanks, Issa I have removed those comments

File: src/controllers/translate.js (hunk 4 above)
Comment (GuevaraMarius) [reply src/controllers/translate.js]: Thanks, Issa I have removed those comments

File: src/controllers/translate.js (hunk 2 above)
Comment (GuevaraMarius) [reply src/controllers/translate.js]: Thanks, @41hulk  I have removed those comments

File: .gitignore (hunk 1 above)
Comment (GuevaraMarius) [reply .gitignore]: Thanks, @41hulk

**Review by KundwaBruno — APPROVED** [review]

File: src/controllers/translate.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+const CREDENTIALS = JSON.parse(process.env.CREDENTIALS);
+
+const translate = new Translate({
+  credentials: CREDENTIALS,
+  projectId: CREDENTIALS.project_id,
+});
+
+const translateText = async (req, res) => {
+  var text = req.body.html;
+  let targetLanguage = req.body.language;
```
Comment (raymond42) [inline src/controllers/translate.js:19]: could you destructure text and targetLanguage from req.body on one line please? and use es6 features

File: src/controllers/bus.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        limit,
+        offset,
+      });
+      const pagination = paginate(page, count, rows, limit);
+  
+      if (offset >= count) {
+        res.status(401).json({
+          message: res.__('There are no buses registered in the system'),
+        });
+        errorRes(res,401,'There are no buses registered in the system')
```
Comment (raymond42) [inline src/controllers/bus.js]: the error messages are not in match with the status codes. And what's the difference between line 32 and line 35?

File: src/controllers/bus.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+export const deleteBus = async (req,res)=>{
+    try {
+      
+        const id=req.params.id;
+        const bus= await  Bus.destroy({
+            where: { id },
+          });
+        if(bus){
+          return successRes(res, 200, 'Bus deleted successfully');
```
Comment (raymond42) [inline src/controllers/bus.js]: could you also return the data containing the bus info that was deleted?

**Review by raymond42 — COMMENTED** [review]
@GuevaraMarius good job on the work done. Please attend to the comments left and revert

**Review by raymond42 — CHANGES_REQUESTED** [review]
@GuevaraMarius good job on the work done. Please attend to the comments left and revert

**Review by samuel776 — COMMENTED** [review]
great Guevera you did good job, i can suggest you to squash those commits into one for best practice.

File: src/controllers/bus.js (hunk 7 above)
Comment (GuevaraMarius) [reply src/controllers/bus.js]: Thanks, the changes have been made

File: .gitignore [hunk 8]
```diff
@@ -2,4 +2,5 @@ node_modules
 .nyc_output
 .env
 build
-.DS_STORE
```
Comment (raymond42) [inline .gitignore:5]: is there a specific reason why you deleted this line?

File: .nycrc [hunk 9]
```diff
@@ -1 +1 @@
-{"all": true,  "include": [    "src/*/**.js"  ],  "exclude": [ "src/controllers/i18n.js" ]}
\ No newline at end of file
+{"all": true,  "include": [    "src/*/**.js"  ],  "exclude": [ "src/controllers/i18n.js","src/controllers/translate.js" ]}
```
Comment (raymond42) [inline .nycrc:1]: is there a specific reason why you made this change?

**Review by raymond42 — CHANGES_REQUESTED** [review]
there is an issue with your rebasing, your code is not updated from what we have on the main branch. Please work on it

**Pull request closed without merging**